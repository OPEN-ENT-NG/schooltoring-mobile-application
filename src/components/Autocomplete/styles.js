import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  input: {
    backgroundColor: COLORS.WHITE
  },
  list: {
    backgroundColor: COLORS.WHITE,
    zIndex: 10,
    position: "relative"
  },
  item: {
    padding: 20
  }
});
