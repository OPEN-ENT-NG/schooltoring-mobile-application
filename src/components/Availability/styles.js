import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    paddingHorizontal: 30,
    paddingVertical: 15,
    flex: 1,
    justifyContent: "space-evenly"
  },
  item: {
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
    height: 60
  },
  itemSwitch: {
    marginRight: 20
  },
  itemText: {
    fontSize: 18
  },
  list: {
    flex: 1,
    justifyContent: "space-around"
  },
  title: {
    fontSize: 18,
    marginBottom: 40,
    flex: 0
  },
  buttonView: {
    flexDirection: "row",
    justifyContent: "flex-end",
    flex: 0,
    padding: 15
  }
});
