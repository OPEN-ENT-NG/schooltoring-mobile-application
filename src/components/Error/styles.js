import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column"
  },
  imageCritical: {
    width: 280,
    height: 301,
    position: "absolute",
    bottom: 0,
    left: -30
  },
  imageUncritical: {
    width: 262,
    height: 301,
    position: "absolute",
    bottom: 0,
    right: -30
  }
});

export default styles;
