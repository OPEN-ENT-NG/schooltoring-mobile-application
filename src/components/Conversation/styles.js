import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

const textColor = {
  color: COLORS.TEXT
};

export default StyleSheet.create({
  conversation: {
    flexDirection: "row",
    paddingHorizontal: 10
  },
  avatar: {
    alignSelf: "flex-start",
    flex: 1,
    alignItems: "center",
    paddingVertical: 20,
    marginHorizontal: 5
  },
  content: {
    flex: 4,
    justifyContent: "space-evenly",
    borderBottomColor: "#000",
    borderBottomWidth: StyleSheet.hairlineWidth
  },
  name: {
    ...textColor,
    fontWeight: "bold",
    fontSize: 20
  },
  message: {
    flexDirection: "row",
    justifyContent: "space-evenly"
  },
  icon: {
    fontSize: 20,
    marginRight: 10
  },
  text: {
    ...textColor,
    flex: 5,
    fontSize: 16,
    flexWrap: "wrap"
  },
  date: {
    flexGrow: 2,
    alignItems: "flex-end"
  },
  dateText: {
    ...textColor
  }
});
