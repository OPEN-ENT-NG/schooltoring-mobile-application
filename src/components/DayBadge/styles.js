import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  item: {
    justifyContent: "center",
    alignItems: "center",
    width: 30,
    height: 30,
    backgroundColor: COLORS.GREY
  },
  itemText: {
    flex: 1,
    color: COLORS.WHITE,
    fontSize: 20
  },
  available: {
    backgroundColor: COLORS.NEGATIVE
  }
});
