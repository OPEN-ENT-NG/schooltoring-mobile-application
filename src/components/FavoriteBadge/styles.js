import { StyleSheet, Platform } from "react-native";

import { COLORS } from "../../styles/common";

const shadow =
  Platform.OS === "android"
    ? {
        elevation: 4
      }
    : {
        shadowColor: COLORS.GREY,
        shadowOffset: { width: 2, height: 2 },
        shadowOpacity: 0.6,
        shadowRadius: 1
      };

export default StyleSheet.create({
  badge: {
    alignItems: "stretch",
    width: "50%",
    padding: 10
  },
  top: {
    aspectRatio: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: COLORS.SECONDARY_BACKGROUND,
    ...shadow
  },
  bottom: {
    backgroundColor: COLORS.BACKGROUND,
    ...shadow,
    maxHeight: 120
  },
  name: {
    textAlign: "center",
    color: COLORS.TEXT,
    fontWeight: "bold",
    fontSize: 16
  },
  buttons: {
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: "center",
    padding: 5
  },
  chatButton: {
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: COLORS.PRIMARY
  },
  loader: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: "rgba(255, 255, 255, 0.8)",
    elevation: 1000,
    zIndex: 5
  }
});
