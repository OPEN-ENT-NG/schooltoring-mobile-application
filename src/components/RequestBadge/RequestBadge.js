import React from "react";
import { View, Text } from "react-native";
import PropTypes from "prop-types";

import Icon from "react-native-vector-icons/MaterialIcons";

import styles from "./styles";
import { COLORS } from "../../styles/common";

import Avatar from "../Avatar/Avatar";
import SubjectBadge from "../SubjectBadge/SubjectBadge";
import SecondaryButton from "../SecondaryButton/SecondaryButton";
import Loader from "../Loader/Loader";
import Touchable from "../Touchable/Touchable";

const getColor = state => {
  return state === "STRENGTH" ? COLORS.SECONDARY : COLORS.PRIMARY;
};

const RequestBadge = props => {
  return (
    <View style={styles.badge}>
      <Touchable style={styles.left} onPress={props.viewProfile}>
        <Avatar
          size={80}
          src={`${global.config.auth.endpoint}${props.userinfo.avatar}`}
        />
      </Touchable>
      <View style={styles.right}>
        <View style={styles.identity}>
          <View style={styles.nameContainer}>
            <Touchable onPress={props.viewProfile}>
              <Text style={styles.name} id="username-field">
                {props.userinfo.username}
              </Text>
            </Touchable>
            <SecondaryButton id="favorite-button" onPress={props.onFavorite}>
              <Icon
                name="favorite"
                color={props.isFavorite ? "red" : COLORS.GREY}
                size={20}
              />
            </SecondaryButton>
          </View>
          <Text style={styles.titleText} id="classname-field">
            {props.userinfo.classNames[0].split("$")[1]}
          </Text>
          <View style={styles.features}>
            {props.subjects.map(subject => (
              <SubjectBadge
                key={subject.subjectId}
                color={getColor(props.state)}
                title={subject.subjectLabel}
              />
            ))}
          </View>
        </View>
        <View style={styles.buttons}>
          <SecondaryButton
            id="refuse-button"
            style={styles.declineButton}
            onPress={props.onRefuse}>
            <Icon name="clear" style={styles.roundButtonIcon} />
          </SecondaryButton>
          <SecondaryButton
            id="accept-button"
            style={styles.acceptButton}
            onPress={props.onAccept}>
            <Icon name="chat" style={styles.roundButtonIcon} />
          </SecondaryButton>
        </View>
      </View>
      {props.loading && (
        <Loader style={styles.loader} color={getColor(props.state)} />
      )}
    </View>
  );
};

RequestBadge.propTypes = {
  userinfo: PropTypes.shape({
    username: PropTypes.string.isRequired,
    classNames: PropTypes.array.isRequired,
    avatar: PropTypes.string
  }).isRequired,
  subjects: PropTypes.arrayOf(
    PropTypes.shape({
      subjectId: PropTypes.string.isRequired,
      subjectLabel: PropTypes.string.isRequired
    })
  ).isRequired,
  loading: PropTypes.bool.isRequired,
  onAccept: PropTypes.func.isRequired,
  onRefuse: PropTypes.func.isRequired,
  onFavorite: PropTypes.func.isRequired,
  isFavorite: PropTypes.bool
};

export default RequestBadge;
