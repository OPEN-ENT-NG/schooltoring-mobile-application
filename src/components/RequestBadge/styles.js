import { StyleSheet, Platform } from "react-native";

import { COLORS } from "../../styles/common";

const roundButton = {
  width: 50,
  height: 50,
  borderRadius: 25
};

const shadowBadge =
  Platform.OS === "android"
    ? {
        elevation: 4
      }
    : {
        shadowColor: COLORS.GREY,
        shadowOffset: { width: 2, height: 2 },
        shadowOpacity: 1,
        shadowRadius: 2
      };

const shadowContent =
  Platform.OS === "android"
    ? {
        elevation: 3
      }
    : {
        shadowColor: COLORS.GREY,
        shadowOffset: { width: -2, height: -1 },
        shadowOpacity: 0.5,
        shadowRadius: 1
      };

const text = { color: COLORS.TEXT };

export default StyleSheet.create({
  badge: {
    flexDirection: "row",
    alignItems: "stretch",
    margin: 20,
    backgroundColor: COLORS.SECONDARY_BACKGROUND,
    flex: 1,
    ...shadowBadge
  },
  features: {
    flexDirection: "row",
    flexWrap: "wrap"
  },
  left: {
    justifyContent: "center",
    alignItems: "flex-start",
    flex: 1
  },
  right: {
    backgroundColor: COLORS.WHITE,
    flexDirection: "row",
    flex: 5,
    padding: 3,
    ...shadowContent
  },

  identity: {
    marginLeft: 15,
    flex: 4,
    justifyContent: "space-evenly"
  },
  name: {
    ...text,
    fontWeight: "bold",
    fontSize: 26
  },
  titleText: {
    ...text,
    fontSize: 20
  },

  buttons: {
    justifyContent: "space-around",
    padding: 15
  },
  declineButton: {
    ...roundButton,
    backgroundColor: COLORS.SECONDARY
  },
  acceptButton: {
    ...roundButton,
    backgroundColor: COLORS.PRIMARY
  },
  roundButtonIcon: {
    fontSize: 25,
    color: COLORS.WHITE
  }
});
