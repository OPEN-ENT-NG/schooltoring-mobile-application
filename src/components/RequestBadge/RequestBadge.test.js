import React from "react";
import { shallow } from "enzyme";

import { COLORS } from "../../styles/common";
import state from "../../jest/state.json";
import RequestBadge from "./RequestBadge";

const request = {
  userinfo: state.userinfo,
  subjects: [
    { subjectId: "1474925-1517929209848", subjectLabel: "FRANCAIS" },
    { subjectId: "1480091-1517929209848", subjectLabel: "ALLEMAND LV2" },
    { subjectId: "1474895-1517929209848", subjectLabel: "ANGLAIS LV1" }
  ]
};

const onAccept = jest.fn();
const onRefuse = jest.fn();

const rendered = shallow(
  <RequestBadge
    userinfo={request.userinfo}
    subjects={request.subjects}
    onAccept={onAccept}
    onRefuse={onRefuse}
  />
);

describe("Render", () => {
  test("RequestBadge match should render without crashing", () =>
    expect(rendered).toMatchSnapshot());

  test(`Avatar source should be '${global.config.auth.endpoint}${
    request.userinfo.avatar
  }'`, () => {
    expect(
      rendered
        .find("Avatar")
        .dive()
        .find("Image")
        .props().source.uri
    ).toEqual(`${global.config.auth.endpoint}${request.userinfo.avatar}`);
  });

  test(`Username and Class name should be respectively '${
    request.userinfo.username
  }' and '${request.userinfo.classNames[0].split("$")[1]}'`, () => {
    const usernameText = rendered.find("Text").find('[id="username-field"]');
    const classnameText = rendered.find("Text").find('[id="classname-field"]');

    expect(usernameText.contains(request.userinfo.username)).toBeTruthy();
    expect(
      classnameText.contains(request.userinfo.classNames[0].split("$")[1])
    ).toBeTruthy();
  });

  test("RequestBadge should render 3 subjects", () =>
    expect(rendered.find("SubjectBadge").length).toEqual(3));

  test("RequestBadge should render 'FRANCAIS', 'ALLEMAND LV2' and 'ANGLAIS LV1' as subjects", () => {
    const features = rendered.find("SubjectBadge");
    expect(
      features
        .at(0)
        .dive()
        .find("Text")
        .contains("FRANCAIS")
    ).toBeTruthy();
    expect(
      features
        .at(1)
        .dive()
        .find("Text")
        .contains("ALLEMAND LV2")
    ).toBeTruthy();
    expect(
      features
        .at(2)
        .dive()
        .find("Text")
        .contains("ANGLAIS LV1")
    ).toBeTruthy();
  });

  test("RequestBadge should render 2 SecondaryButtons", () =>
    expect(rendered.find("SecondaryButton").length).toEqual(2));

  test("RequestBadge should render 2 buttons with respectively 'clear' and 'chat' as Icon", () => {
    const buttons = rendered.find("SecondaryButton");

    expect(
      buttons
        .at(0)
        .children()
        .find("Icon")
        .props().name
    ).toEqual("clear");
    expect(
      buttons
        .at(1)
        .children()
        .find("Icon")
        .props().name
    ).toEqual("chat");
  });
});

describe("Interactions", () => {
  test("On press on clear button should call onRefuse prop function", () => {
    onRefuse.mockReset();
    const onRefuseButton = rendered.find("SecondaryButton").at(0);
    onRefuseButton.props().onPress();

    expect(onRefuse).toHaveBeenCalled();
  });

  test("On press on chat button should call onAccept prop function", () => {
    onAccept.mockReset();
    const onAcceptButton = rendered.find("SecondaryButton").at(1);
    onAcceptButton.props().onPress();

    expect(onAccept).toHaveBeenCalled();
  });
});
