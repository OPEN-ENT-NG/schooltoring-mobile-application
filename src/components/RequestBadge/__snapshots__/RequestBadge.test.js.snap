// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RequestBadge match should render without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#e3f2fd",
      "flex": 1,
      "flexDirection": "row",
      "margin": 20,
      "shadowColor": "#929292",
      "shadowOffset": Object {
        "height": 2,
        "width": 2,
      },
      "shadowOpacity": 1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Avatar
      size={80}
      src="http://ent3-core.gdapublic.fr/workspace/document/49a54de3-51f8-464b-81bb-a377f508f382"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 5,
        "flexDirection": "row",
        "padding": 3,
        "shadowColor": "#929292",
        "shadowOffset": Object {
          "height": -1,
          "width": -2,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "justifyContent": "space-evenly",
          "marginLeft": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          id="username-field"
          style={
            Object {
              "color": "#232323",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          John DOE
        </Text>
        <SecondaryButton
          id="favorite-button"
          onPress={[MockFunction]}
        >
          <Icon
            allowFontScaling={false}
            color="red"
            name="favorite"
            size={20}
          />
        </SecondaryButton>
      </View>
      <Text
        id="classname-field"
        style={
          Object {
            "color": "#232323",
            "fontSize": 16,
          }
        }
      >
        2DE4
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <SubjectBadge
          color="#17b6b0"
          key="1474925-1517929209848"
          title="FRANCAIS"
        />
        <SubjectBadge
          color="#17b6b0"
          key="1480091-1517929209848"
          title="ALLEMAND LV2"
        />
        <SubjectBadge
          color="#17b6b0"
          key="1474895-1517929209848"
          title="ANGLAIS LV1"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "justifyContent": "space-around",
          "padding": 15,
        }
      }
    >
      <SecondaryButton
        id="refuse-button"
        onPress={[MockFunction]}
        style={
          Object {
            "backgroundColor": "#e91e63",
            "borderRadius": 25,
            "height": 50,
            "margin": 5,
            "width": 50,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          name="clear"
          size={12}
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 25,
            }
          }
        />
      </SecondaryButton>
      <SecondaryButton
        id="accept-button"
        onPress={[MockFunction]}
        style={
          Object {
            "backgroundColor": "#17b6b0",
            "borderRadius": 25,
            "height": 50,
            "margin": 5,
            "width": 50,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          name="chat"
          size={12}
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 25,
            }
          }
        />
      </SecondaryButton>
    </View>
  </View>
</View>
`;
